src/formats/AmberNetCDF.cpp

```diff
@@ -246,38 +326,83 @@ void AmberNetCDFFormat::write(const Frame& frame) {
     step_++;
 }
 
-void AmberNetCDFFormat::write_array(const std::vector<Vector3D>& array, const std::string& name) {
-    auto var = file_.variable<nc::NcFloat>(name);
+template <AmberFormat F>
+void Amber<F>::write_array(const std::vector<Vector3D>& array, const std::string& name) {
+    auto var = get_variable<F>(file_, name);
     auto natoms = array.size();
-    std::vector<size_t> start{step_, 0, 0};
-    std::vector<size_t> count{1, natoms, 3};
+    auto range = vec3d_n_range(natoms);
 
-    auto data = std::vector<float>(natoms * 3);
+    auto data = std::vector<typename FormatSpec<F>::type>(natoms * 3);
```

Comment:
This is alright as-is, you could also add `using real_t = typename FormatSpec<F>::type` at the begining of the function and then use `real_t`  everywhere

Reply:
That's much easier to the eye. Done.